pkg/tuning/preset-tuning.go

```diff
@@ -129,10 +162,55 @@ func CreatePresetTuning(ctx context.Context, workspaceObj *kaitov1alpha1.Workspa
 		volumeMounts = append(volumeMounts, shmVolumeMount)
 	}
 
+	// Add shared volume for tuning parameters
 	cmVolume, cmVolumeMount := utils.ConfigCMVolume(workspaceObj.Tuning.ConfigTemplate)
 	volumes = append(volumes, cmVolume)
 	volumeMounts = append(volumeMounts, cmVolumeMount)
 
+	// Add shared volume for results dir
+	resultsVolume, resultsVolumeMount := utils.ConfigResultsVolume()
+	if resultsVolume.Name != "" {
+		volumes = append(volumes, resultsVolume)
+	}
+	if resultsVolumeMount.Name != "" {
+		volumeMounts = append(volumeMounts, resultsVolumeMount)
+	}
+	return volumes, volumeMounts
+}
+
+func CreatePresetTuning(ctx context.Context, workspaceObj *kaitov1alpha1.Workspace,
+	tuningObj *model.PresetParam, kubeClient client.Client) (client.Object, error) {
+	var initContainers, sidecarContainers []corev1.Container
+	volumes, volumeMounts := setupDefaultSharedVolumes(workspaceObj)
+
+	initContainer, imagePullSecrets, dataSourceVolume, dataSourceVolumeMount, err := prepareDataSource(ctx, workspaceObj)
+	if err != nil {
+		return nil, err
+	}
+	volumes = append(volumes, dataSourceVolume)
+	volumeMounts = append(volumeMounts, dataSourceVolumeMount)
+	if initContainer.Name != "" {
+		initContainers = append(initContainers, initContainer)
+	}
+
+	sidecarContainer, imagePushSecret, dataDestVolume, dataDestVolumeMount, err := prepareDataDestination(ctx, workspaceObj)
+	if err != nil {
+		return nil, err
+	}
+	volumes = append(volumes, dataDestVolume)
+	volumeMounts = append(volumeMounts, dataDestVolumeMount)
+	if sidecarContainer.Name != "" {
```

Comment:
I still prefer a nil check over checking for empty string. sidecarContainer can be defined as a pointer type.
The same for imagePushSecret.

Reply:
fixed!